graphix/pyzx.py

```
@@ -0,0 +1,164 @@
+    # the PyZX graph and vice versa.
+    map_to_og = dict(zip(body_verts, og._inside.nodes()))
+    map_to_pyzx = {v: i for i, v in map_to_og.items()}
+
+    # Open Graph's don't have boundary nodes, so we need to connect the
+    # input and output Z spiders to their corresponding boundary nodes in
+    # pyzx.
+    for pyzx_index, og_index in zip(in_verts, og.inputs):
+        g.add_edge((pyzx_index, map_to_pyzx[og_index]))
+    for pyzx_index, og_index in zip(out_verts, og.outputs):
+        g.add_edge((pyzx_index, map_to_pyzx[og_index]))
+
+    og_edges = og._inside.edges()
+    pyzx_edges = [(map_to_pyzx[a], map_to_pyzx[b]) for a, b in og_edges]
+    g.add_edges(pyzx_edges, zx.EdgeType.HADAMARD)
+
+    # Add the edges between the Z spiders in the graph body
+    for og_index, meas in og.measurements.items():
+        # If it's an X measured node, then we handle it in the next loop
+        if meas.plane == Plane.XY:
+            g.set_phase(map_to_pyzx[og_index], meas.angle)
+
+    # Connect the X measured vertices
+    for og_index, pyzx_index in zip(x_meas, x_meas_verts):
+        g.add_edge((map_to_pyzx[og_index], pyzx_index), zx.EdgeType.HADAMARD)
+        g.set_phase(pyzx_index, og.measurements[og_index].angle)
+
+    return g
+
+
+def from_pyzx_graph(g: zx.graph.base.BaseGraph) -> OpenGraph:
+    """Constructs an Optyx Open Graph from a PyZX graph.
+
+    This method may add additional nodes to the graph so that it adheres
+    with the definition of an OpenGraph. For instance, if the final node on
+    a qubit is measured, it will add two nodes behind it so that no output
+    nodes are measured to satisfy the requirements of an open graph.
+
+    Example
+    -------
+    >>> import pyzx as zx
+    >>> from graphix.opengraph import OpenGraph
+    >>> circ = zx.qasm("qreg q[2]; h q[1]; cx q[0], q[1]; h q[1];")
+    >>> g = circ.to_graph()
+    >>> og = OpenGraph.from_pyzx_graph(g)
+    """
+    zx.simplify.to_graph_like(g)
```

```suggestion
    # check pyzx availability and version
    try:
        import pyzx as zx
    except ModuleNotFoundError as e:
        msg = "Cannot find pyzx (optional dependency)."
        raise RuntimeError(msg) from e
    if zx.__version__ != "0.8.0":
        raise Warning(
            "`from_pyzx_graph` is guaranteed to work only with pyzx==0.8.0 due to possible breaking changes in `pyzx`."
        )
        
    zx.simplify.to_graph_like(g)
```

---

`raise` makes python abort (rather than warning)

---

ah thanks, yes - changing now